src/use-swr.ts

```diff
@@ -513,6 +518,14 @@ function useSWR<Data = any, Error = any>(
           revalidators.pop()
         }
       }
+      if (onReconnect && RECONNECT_REVALIDATORS[key]) {
```

Comment:
the recycle revalidator part is kind of repetitive, we can re-abstract them but I feel it would be good in the separate PR